pkg/vpcmodel/semanticDiff.go

```diff
@@ -29,64 +36,100 @@ type connectionDiff struct {
 	diff  DiffType
 }
 
-type SubnetsDiff map[VPCResourceIntf]map[VPCResourceIntf]*connectionDiff
+type connectivesDiff map[VPCResourceIntf]map[VPCResourceIntf]*connectionDiff
 
 type ConfigsForDiff struct {
-	config1 *VPCConfig
-	config2 *VPCConfig
+	config1      *VPCConfig
+	config2      *VPCConfig
+	diffAnalysis diffAnalysisType
 }
 
-type SubnetConfigConnectivity struct {
-	config             *VPCConfig
-	subnetConnectivity SubnetConnectivityMap
+type configConnectivity struct {
+	config       *VPCConfig
+	connectivity generalConnectivityMap
 }
 
-type DiffBetweenSubnets struct {
-	subnet1Subtract2 SubnetsDiff
-	subnet2Subtract1 SubnetsDiff
+type diffBetweenCfgs struct {
+	cfg1ConnRemovedFrom2 connectivesDiff
+	cfg2ConnRemovedFrom1 connectivesDiff
+	diffAnalysis         diffAnalysisType
 }
 
-func (configs ConfigsForDiff) GetSubnetsDiff(grouping bool) (*DiffBetweenSubnets, error) {
-	// 1. compute connectivity for each of the subnets
-	subnetsConn1, err := configs.config1.GetSubnetsConnectivity(true, grouping)
-	if err != nil {
-		return nil, err
-	}
-	subnetsConn2, err := configs.config2.GetSubnetsConnectivity(true, grouping)
-	if err != nil {
-		return nil, err
+func (configs ConfigsForDiff) GetDiff(grouping bool) (*diffBetweenCfgs, error) {
```

Comment:
exposed functions should have documentation

Reply:
really ;-) ?
Added

Reply:
And also exposed types.